eth/downloader/peer.go

```diff
@@ -409,7 +419,12 @@ func (ps *peerSet) BodyIdlePeers() ([]*peer, int) {
 	idle := func(p *peer) bool {
 		return atomic.LoadInt32(&p.blockIdle) == 0
 	}
-	return ps.idlePeers(62, 64, idle)
+	throughput := func(p *peer) float64 {
```

Comment:
There is ways to avoid it, without reflection. For example, a function could be created which takes a peer and what field to access, and does that + the locking + calling `idlePeers`

I don't understand "it's safer coding this way as we can use the accessor methods any way seen fit and don't need to worry about synchronisation" I'm not talking about changing any functionality here - simply about refactoring to avoid duplicated code. Same goes for the comment about comparison of throughput values and deadlocking - refactoring to avoid duplication here does not have to change the functionality / locking logic.